### PR TITLE
[WIP] Add OSF provider

### PR DESCRIPTION
**Pull request by erinspace**
Includes @fabianvf's JSON transformer classes

### CODE REVIEW

**Comment by fabianvf** [comment]
Can you bring this up to date with develop?

File: scrapi/harvesters/osf.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+class OSFHarvester(JSONHarvester):
+    short_name = 'osf'
+    long_name = 'Open Science Framework'
+    url = 'http://osf.io/api/v1/search/'
+    count = 0
+
+    # Only registrations that aren't just the word "test" or "test project"
+    # TODO - change this url to production once in production
+    URL = 'https://staging.osf.io/api/v1/search/?q=category:registration ' +\
```
Comment (fabianvf) [inline scrapi/harvesters/osf.py]: Change this to production, I won't merge this until the OSF changes are live.

**Comment by erinspace** [comment]
Yep, since the API hasn't been updated in prod yet, tho it is updated on develop...

File: scrapi/harvesters/osf.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    # Only registrations that aren't just the word "test" or "test project"
+    URL = 'https://staging.osf.io/api/v1/search/?q=category:registration ' +\
+          ' AND date_created:[{} TO {}]' +\
+          ' AND NOT title=test AND NOT title="Test Project"&size=1000'
+
+    @property
+    def schema(self):
+        return {
+            'contributor': ('contributors', process_contributors),
```
Comment (fabianvf) [inline scrapi/harvesters/osf.py]: I think you'll need to bring this PR up to date with the new schema again

**Pull request merged**